src/openms/source/CHEMISTRY/ModificationsDB.cpp

```diff
@@ -50,10 +50,16 @@ using namespace std;
 
 namespace OpenMS
 {
-  ModificationsDB::ModificationsDB()
+  ModificationsDB::ModificationsDB(OpenMS::String unimod_file, OpenMS::String psimod_file)
   {
-    readFromUnimodXMLFile("CHEMISTRY/unimod.xml");
-    readFromOBOFile("CHEMISTRY/PSI-MOD.obo");
+    if (!unimod_file.empty())
```

Comment:
that wont work as you then never really call `readFromOBOFile` *at all* if I see this correctly. I think therefore your code can lead to a situation where either the unimod or obo file is never read which is quite dangerous